blenderproc/python/writer/BopWriterUtility.py

```diff
@@ -614,16 +655,28 @@ def calc_gt_info(chunk_dirs: List[str], dataset_objects: List[MeshObject], start
                 K = np.array(scene_camera[im_id]['cam_K']).reshape(3, 3)
                 fx, fy, cx, cy = K[0, 0], K[1, 1], K[0, 2], K[1, 2]
                 im_size = (depth.shape[1], depth.shape[0])
+                camera = pyrender.IntrinsicsCamera(fx=fx, fy=fy, cx=cx+ren_cx_offset, cy=cy+ren_cy_offset)
 
                 scene_gt_info[im_id] = []
                 for gt in scene_gt[im_id]:
-                    # Render depth image of the object model in the ground-truth pose.
-                    depth_gt_large = ren.render_object(
-                        gt['obj_id'], np.array(gt['cam_R_m2c']).reshape(3, 3), np.array(gt['cam_t_m2c']),
-                        fx, fy, cx + ren_cx_offset, cy + ren_cy_offset)['depth']
+                    # create a new scene
+                    scene = pyrender.Scene()
+
+                    # add camera and current object
+                    scene.add(camera)
+                    t = np.array(gt['cam_t_m2c'])
+                    if m2mm:  # convert back to meters in case they have been saved in mm
+                        t /= 1000.
+                    pose = bop_pose_to_pyrender_coordinate_system(cam_R_m2c=np.array(gt['cam_R_m2c']).reshape(3, 3),
+                                                                  cam_t_m2c=t)
+                    scene.add(dataset_objects[gt['obj_id']], pose=pose)
+
+                    # render the depth image
+                    _, depth_gt_large = renderer.render(scene=scene)
                     depth_gt = depth_gt_large[
                                ren_cy_offset:(ren_cy_offset + im_height),
                                ren_cx_offset:(ren_cx_offset + im_width)]
+                    depth_gt *= 1000.  # convert to mm
```

Comment:
Shouldnt there also be an if around that or is depth always stored in mm?

Reply:
The blenderproc generated depth is always changed to mm [here](https://github.com/DLR-RM/BlenderProc/blob/main/blenderproc/python/writer/BopWriterUtility.py#L531)